arch/arc/core/mpu/Kconfig

```diff
@@ -28,6 +28,7 @@ config ARC_MPU
 	select ARC_CORE_MPU
 	select THREAD_STACK_INFO
 	select MEMORY_PROTECTION
+	select GEN_PRIV_STACKS
```

Comment:
shouldn't we only be doing this if ARC_MPU_VER = 2?
Not seeing benefit for later MPU revisions which don't have power-of-two alignment constraints.

Reply:
Note: ARM doesn't do this correctly either. There is an open issue #24048

Reply:
I know GEN_PRIV_STACKS is not necessary for MPU_VER 3.  Enable it for V3 is to have the same logic for MPU drivers to simplify the code.  @ruuddw  @abrodkin , what's your opinion. If required ,  I can keep the original logic for MPU_VER 3

Reply:
I add a commit to disable GEN_PRIV_STACK for ARC MPU v3